### PR TITLE
test: fix intermittent failure in p2p_getaddr_caching.py

### LINKED ISSUE #28133: p2p_getaddr_caching.py failure in TSan CI
https://cirrus-ci.com/task/5471865133596672?logs=ci#L3981

```
 test  2023-07-22T19:16:17.696000Z TestFramework (ERROR): Assertion failed 
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 131, in main
                                       self.run_test()
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_getaddr_caching.py", line 74, in run_test
                                       addr_receiver_onion1 = self.nodes[0].add_p2p_connection(AddrReceiver(), dstport=self.onion_port1)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_node.py", line 652, in add_p2p_connection
                                       assert_equal(len(info), 1)
                                     File "/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 57, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(2 == 1)
 test  2023-07-22T19:16:17.697000Z TestFramework (DEBUG): Closing down network thread 
 test  2023-07-22T19:16:17.747000Z TestFramework (INFO): Stopping nodes 
 test  2023-07-22T19:16:17.748000Z TestFramework.node0 (DEBUG): Stopping node 
```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by mzumsande**
Fixes #28133

In the consistency check, it's not enough to check that our address/port is unique, only the combination of source and target must be unique. Otherwise, the OS may reuse ports for connections to different `-addrbind`, which was happening in the failed runs.

While at it, the second commit cleans up duplicate `getaddr` messages in `p2p_getaddr_caching.py` that do nothing but generate `Ignoring repeated "getaddr"` log messages (and cleans up some whitespace the python linter complains about).

### CODE REVIEW

File: test/functional/p2p_getaddr_caching.py [hunk 1]
```diff
@@ -70,11 +71,8 @@ def run_test(self):
         cur_mock_time = int(time.time())
         for i in range(N):
             addr_receiver_local = self.nodes[0].add_p2p_connection(AddrReceiver())
-            addr_receiver_local.send_and_ping(msg_getaddr())
```
Comment (vasild) [inline test/functional/p2p_getaddr_caching.py:73]: There is a similar pattern below on lines 108,110,112. Should be possible to remove those too? If yes, then also `from test_framework.messages import msg_getaddr`.

To other reviewers: the `GETADDR` message is sent at the end of `P2PInterface::on_version()`.

File: test/functional/test_framework/test_node.py [hunk 2]
```diff
@@ -645,10 +645,11 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
             p2p_conn.sync_with_ping()
 
             # Consistency check that the node received our user agent string.
-            # Find our connection in getpeerinfo by our address:port, as it is unique.
+            # Find our connection in getpeerinfo by our address:port and theirs, as this combination is unique.
             sockname = p2p_conn._transport.get_extra_info("socket").getsockname()
             our_addr_and_port = f"{sockname[0]}:{sockname[1]}"
-            info = [peer for peer in self.getpeerinfo() if peer["addr"] == our_addr_and_port]
+            dst_addr_and_port =f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```
Comment (vasild) [inline test/functional/test_framework/test_node.py:651]: nit, missing white space after `=`:
```suggestion
            dst_addr_and_port = f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```

**Review by vasild — APPROVED** [review]
ACK e4d37875e6a440748623250d77ec3be523a1d34d

Today I learned something new: it is possible to have duplicate (srcaddr, srcport) for TCP connections if the destination is different. Thank you!

File: test/functional/p2p_getaddr_caching.py (hunk 1 above)
Comment (mzumsande) [reply test/functional/p2p_getaddr_caching.py:73]: yes, I missed those, now I removed them all!

File: test/functional/test_framework/test_node.py (hunk 2 above)
Comment (mzumsande) [reply test/functional/test_framework/test_node.py]: fixed

**Comment by mzumsande** [comment]
[e4d3787 ](https://github.com/bitcoin/bitcoin/commit/e4d37875e6a440748623250d77ec3be523a1d34d)to 8a20f76: Addressed feedback, thanks!

> Today I learned something new: it is possible to have duplicate (srcaddr, srcport) for TCP connections if the destination is different. Thank you!

I was also surprised by this!

**Review by vasild — APPROVED** [review]
ACK 8a20f765cce2fc0fadf1a2b66b843b67f2d2ae12

**Pull request merged**